CONTRIBUTING.md

```diff
@@ -1,60 +1,145 @@
-# Typical development workflow
+# Contributing
 
-Fork the repo, work on an issue
+This guide explains how to set up your environment, make changes, and submit a PR.
 
-## Updating the generated Kibana client.
+## Development Setup
 
-If your work involves the Kibana API, the endpoints may or may not be included in the generated client.
-Check [generated/kbapi](./generated/kbapi/) for more details.
+* Fork and clone the repo.
+* Setup your preferred IDE (Intelliji, VSCode, etc.)
 
-## Acceptance tests
+Requirements:
+* [Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
+* [Go](https://golang.org/doc/install) >= 1.25
+* Docker (for acceptance tests)
 
-```bash
-make docker-testacc
-```
+## Development Workflow
 
-Run a single test with terraform debug enabled:
-```bash
-env TF_LOG=DEBUG make docker-testacc TESTARGS='-run ^TestAccResourceDataStreamLifecycle$$'
-```
+* Create a new branch for your changes.
+* Make your changes. See [Useful Commands](#useful-commands) and [Debugging](#debugging).
+  * GitHub Copilot can be used to help with these changes (See [Using Copilot](#using-copilot))
+* Validate your changes
+  * Run unit and acceptance tests (See [Running Acceptance Tests](#running-acceptance-tests)).
+  * Run `make fmt`, `make lint`.
+  * All checks also run automatically on every PR.
+* Add a changelog entry in `CHANGELOG.md` under the `Unreleased` section. This will be included in the release notes of the next release.
+* Submit your PR for review.
```

Comment:
Currently changelog entries link back to the PR introducing the change which means we first need to open the PR and then add the changelog entry. That situation has never really made sense, but it does make sense to be consistent for now. 

IMO rather than improving the ordering here, it would likely make more sense to look at automatically adding changelog entries based on the PR title.

Reply:
I'll reorder it for now. Though it would be nice to have this automated so the release job just collects release notes from all PRs included in the release.